popup/js/popup.js

```diff
@@ -192,35 +200,47 @@ function suggestion() {
 var theme;
 
 function themeChange() {
-
     theme = localStorage.getItem("theme");
-    if(theme == 'light' )
-        localStorage.setItem( "theme" , "dark" );
-    else
-        localStorage.setItem("theme", "light");
+    if (theme === "light") {
+        localStorage.setItem("theme", "dark");
+    } else { localStorage.setItem("theme", "light"); }
 
     var bg = document.getElementById("content");
     bg.classList.toggle("dark");
 
-    var labels = document.getElementsByTagName('label');
+    var labels = document.getElementsByTagName("label");
     labels[0].classList.toggle("dark-label");
     labels[1].classList.toggle("dark-label");
     labels[2].classList.toggle("dark-label");
+}
 
+function recordSearchHistory() {
+    var recentSearchQueries = [];
+    if (localStorage.getItem("search")) {
+        recentSearchQueries = JSON.parse(localStorage.getItem("search"));
+    }
+    var x = text.value;
```

Comment:
Again what is x. Imagine someone coming to read your code and straightaway come to this line and tries to figure out what does x mean.

Also this is an perfect example of how important variable naming is. This `text` variable is being used throughout the file, it was declared in the top part of file. I had to go through the whole file to understand what this text variable points to. If it had a better name it would have been much easier to understand without scrolling hundereds of lines. Also, imagine tomorrow we have 2-3 more text fields. Now i cannot name all of them text (i hope you got the idea we wont be naming them text1 text2 text3 etc.). Then i would have to rename this `text` variable and all places it is used at. Sad.

Most probably it would have been done by me only when i created this project. I learned all of this after looking how other people code stuff. So lets not fix this `text` var but you can fix your var name.